appfl/algorithm/fedavg.py

```diff
@@ -4,71 +4,97 @@
 
 from collections import OrderedDict
 from .algorithm import BaseServer, BaseClient
+from .misc import *
 
 import torch
+from torch.optim import *
 from torch.nn import CrossEntropyLoss
 from torch.utils.data import DataLoader
+import copy
+
 
 class FedAvgServer(BaseServer):
-    def __init__(self, model, num_clients, device, **kwargs):
-        super(FedAvgServer, self).__init__(model, num_clients, device)
+    def __init__(self, weights, model, num_clients, device, **kwargs):
+        super(FedAvgServer, self).__init__(weights, model, num_clients, device)
 
         self.__dict__.update(kwargs)
+        self.num_clients = num_clients
+        self.weights = weights
+
+        self.primal_states = OrderedDict()
+        for i in range(num_clients):
+            self.primal_states[i] = OrderedDict()
```

Comment:
These variables seem used in other algorithm derivations. Any reason that they are defined locally? not in `BaseServer`?